library/tests/src/test_arithmetic.rs

```diff
@@ -0,0 +1,401 @@
+                Microsoft.Quantum.Arithmetic.ApplyXorInPlace(3, a);
+            }
+            apply {
+                set result = [M(a[0]),M(a[1]),M(a[2])];
+            }
+            return result;
+        }"
+        },
+        &Value::Array(vec![Value::RESULT_ONE, Value::RESULT_ONE, Value::RESULT_ZERO].into()),
+    );
+}
+
+#[test]
+fn check_apply_xor_in_place_l() {
+    test_expression(
+        {
+            "{
+            open Microsoft.Quantum.Arithmetic;
+            use q = Qubit[100];
+            ApplyXorInPlaceL(953L <<< 50, q);
+            let result = MeasureInteger(q[50...]);
+            ResetAll(q);
+            result
+        }"
+        },
+        &Value::Int(953),
+    );
+}
+
+#[test]
+fn check_measure_integer() {
+    test_expression(
+        {
+            "{
+                open Microsoft.Quantum.Arithmetic;
+                use q = Qubit[16];
+                ApplyXorInPlace(45967, q);
+                let result = MeasureInteger(q);
+                ResetAll(q);
+                return result;
+            }"
+        },
+        &Value::Int(45967),
+    );
+}
+
+#[test]
+fn check_add_i_nc() {
+    test_expression(
+        {
+            "{  // RippleCarryAdderNoCarryTTK case
+                use x = Qubit[4];
+                use y = Qubit[4];
+                open Microsoft.Quantum.Arithmetic;
+                ApplyXorInPlace(3, x);
+                ApplyXorInPlace(5, y);
+                AddI(x,y); // 3+5=8
+                let result = [M(y[0]),M(y[1]),M(y[2]),M(y[3])];
+                ResetAll(x+y);
+                return result;
+        }"
+        },
+        &Value::Array(
+            vec![
+                Value::RESULT_ZERO,
+                Value::RESULT_ZERO,
+                Value::RESULT_ZERO,
+                Value::RESULT_ONE, // 3+5=8
+            ]
+            .into(),
+        ),
+    );
+}
+
+#[test]
+fn check_add_i_c() {
+    test_expression(
+        {
+            "{  // RippleCarryAdderTTK case
+                use x = Qubit[4];
+                use y = Qubit[5];
+                open Microsoft.Quantum.Arithmetic;
+                ApplyXorInPlace(7, x);
+                ApplyXorInPlace(11, y);
+                AddI(x,y); // 7+11=18
+                let result = [M(y[0]),M(y[1]),M(y[2]),M(y[3]),M(y[4])];
+                ResetAll(x+y);
+                return result;
+        }"
+        },
+        &Value::Array(
+            vec![
+                Value::RESULT_ZERO,
+                Value::RESULT_ONE, // 2
+                Value::RESULT_ZERO,
+                Value::RESULT_ZERO,
+                Value::RESULT_ONE, // 16
+            ]
+            .into(),
+        ), // 10010b = 18
+    );
+}
+
+#[test]
+fn check_add_i_1_1() {
+    test_expression(
+        {
+            "{  // Shortest case
+                use x = Qubit[1];
+                use y = Qubit[1];
+                open Microsoft.Quantum.Arithmetic;
+                X(x[0]);
+                AddI(x,y);
```

Comment:
Will AddI point to IncByLE eventually?

Reply:
We should remove AddI along with the backing code. We have a way of pointing users to the new function if they use the old one.